client/js/widgets/widget.js

```diff
@@ -1656,6 +1664,25 @@ export class Widget extends StateManaged {
     }
   }
 
+  inheritSeatVisibility(seatVisibility) {
+    if (this.get('hoverInheritVisibleForSeat')) {
+      const widgetSeatVisibility = this.get('onlyVisibleForSeat');
+      if (widgetSeatVisibility) {
+        // Filter seatVisibility by current widgets seats.
+        if (!seatVisibility) {
+          seatVisibility = widgetSeatVisibility;
+        } else {
+          let filterTo = new Set(asArray(widgetSeatVisibility));
+          seatVisibility = asArray(seatVisibility).filter((seatId) => { return filterTo.has(seatId); });
+        }
+      }
+    }
+    const thisParent = this.get('parent');
+    if (thisParent)
+      seatVisibility = widgets.get(thisParent).inheritSeatVisibility(seatVisibility);
```

Comment:
This could crash if `parent` property is set to an invalid widget id. Should change the condition to
```
if (thisParent && widgets.has(thisParent))
```

Reply:
Good catch! Fixed here and in updatePiles.

Reply:
Also added an invalid parent in the pr-test to verify.